ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js

```diff
@@ -0,0 +1,266 @@
+import React, { useContext, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import Button from '../../../components/ui/button';
+import PulseLoader from '../../../components/ui/pulse-loader';
+import { CUSTODY_ACCOUNT_ROUTE } from '../../../helpers/constants/routes';
+import {
+  AlignItems,
+  DISPLAY,
+  TextColor,
+  TEXT_ALIGN,
+  FLEX_DIRECTION,
+} from '../../../helpers/constants/design-system';
+import { BUILT_IN_NETWORKS } from '../../../../shared/constants/network';
+import { I18nContext } from '../../../contexts/i18n';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { setProviderType } from '../../../store/actions';
+import { mmiActionsFactory } from '../../../store/institutional/institution-background';
+import { Label, Text } from '../../../components/component-library';
+import Box from '../../../components/ui/box';
+
+const ConfirmAddCustodianToken = () => {
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const trackEvent = useContext(MetaMetricsContext);
+  const mmiActions = mmiActionsFactory();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const connectRequests = useSelector(
+    (state) => state.metamask.institutionalFeatures?.connectRequests,
+  );
+  const complianceActivated = useSelector((state) =>
+    Boolean(state.metamask.institutionalFeatures?.complianceProjectId),
+  );
+  const [showMore, setShowMore] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [connectError, setConnectError] = useState('');
+
+  const handleConnectError = (e) => {
+    let errorMessage = e.message;
+
+    if (!errorMessage) {
+      errorMessage = 'Connection error';
+    }
+
+    setConnectError(errorMessage);
+    setIsLoading(false);
+  };
+
+  const renderSelectedToken = () => {
+    const connectRequest = connectRequests ? connectRequests[0] : undefined;
+
+    return (
+      <Box
+        paddingTop={2}
+        paddingBottom={2}
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.ROW}
+        alignItems={AlignItems.center}
+      >
+        <Text>
+          {showMore && connectRequest?.token
+            ? connectRequest?.token
+            : `...${connectRequest?.token.slice(-9)}`}
+        </Text>
+        {!showMore && (
+          <Box paddingLeft={2}>
+            <a
+              rel="noopener noreferrer"
+              onClick={() => {
+                setShowMore(true);
+              }}
+            >
+              {t('ShowMore')}
+            </a>
```

Comment:
Could use `ButtonLink` here https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-buttonlink--default-story accepts all box props so you may be abled to do `<ButtonLink marginLeft={2} `

Reply:
Ok 👍🏼